zstacklib/zstacklib/utils/version.py

```
@@ -0,0 +1,25 @@
+    e.g., '5.2.0-8.el8' -> (5, 2, 0, 8, 8), '2.a.1' -> (2, 1)
+    """
+    def __init__(self, vstring):
+        self.vstring = str(vstring)
+        self.version = tuple(int(x) for x in re.findall(r'\d+', self.vstring))
+
+    def _pad(self, other):
+        maxlen = max(len(self.version), len(other.version))
+        return (self.version + (0,) * (maxlen - len(self.version)),
+                other.version + (0,) * (maxlen - len(other.version)))
+
+    def __lt__(self, other): a, b = self._pad(other); return a < b
```

Ruff (0.15.7)

[error] Multiple statements on one line (semicolon)

(E702)